slsim/lens_pop.py

```diff
@@ -328,8 +352,19 @@ def __init__(
         else:
             raise ValueError("source_type %s is not supported" % source_type)
         self.cosmo = cosmo
-        self.f_sky = sky_area
-
+        self.source_sky_area = source_sky_area
+        self.deflector_sky_area = deflector_sky_area
+        self.f_sky = full_sky_area
+        if full_sky_area is None:
```

Comment:
I would change that, such that first the sky areas are set in a certain procedure, and then the factors computed.